src/main/java/org/jabref/model/strings/LatexToUnicodeAdapter.java

```diff
@@ -5,23 +5,56 @@
 import java.util.regex.Pattern;
 
 import com.github.tomtung.latex2unicode.LaTeX2Unicode;
+import fastparse.core.Parsed;
 
 /**
  * Adapter class for the latex2unicode lib. This is an alternative to our LatexToUnicode class
  */
 public class LatexToUnicodeAdapter {
 
-    private static Pattern underscoreMatcher = Pattern.compile("_(?!\\{)");
+    private static final Pattern UNDERSCORE_MATCHER = Pattern.compile("_(?!\\{)");
 
-    private static String replacementChar = "\uFFFD";
+    private static final String REPLACEMENT_CHAR = "\uFFFD";
 
-    private static Pattern underscorePlaceholderMatcher = Pattern.compile(replacementChar);
+    private static final Pattern UNDERSCORE_PLACEHOLDER_MATCHER = Pattern.compile(REPLACEMENT_CHAR);
 
+    /**
+     * Attempts to resolve all LaTeX in the String.
+     *
+     * @param inField a String containing LaTeX
+     * @return a String with LaTeX resolved into Unicode, or the original String if the LaTeX could not be parsed
+     */
     public static String format(String inField) {
         Objects.requireNonNull(inField);
 
-        String toFormat = underscoreMatcher.matcher(inField).replaceAll(replacementChar);
-        toFormat = Normalizer.normalize(LaTeX2Unicode.convert(toFormat), Normalizer.Form.NFC);
-        return underscorePlaceholderMatcher.matcher(toFormat).replaceAll("_");
+        try {
+            return parse(inField);
+        } catch (IllegalArgumentException ignored) {
+            return Normalizer.normalize(inField, Normalizer.Form.NFC);
+        }
+    }
+
+    /**
+     * Attempts to resolve all LaTeX in the String.
+     *
+     * @param inField a String containing LaTeX
+     * @return a String with LaTeX resolved into Unicode
+     * @throws IllegalArgumentException if the LaTeX could not be parsed
+     */
+    public static String parse(String inField) throws IllegalArgumentException {
+        Objects.requireNonNull(inField);
+        String toFormat = UNDERSCORE_MATCHER.matcher(inField).replaceAll(REPLACEMENT_CHAR);
+        try {
+            var parsingResult = LaTeX2Unicode.parse(toFormat);
+            if (parsingResult instanceof Parsed.Success) {
+                String text = parsingResult.get().value();
+                toFormat = Normalizer.normalize(text, Normalizer.Form.NFC);
+                return UNDERSCORE_PLACEHOLDER_MATCHER.matcher(toFormat).replaceAll("_");
+            } else {
+                throw new IllegalArgumentException("Parsing of latex failed.");
+            }
+        } catch (Throwable throwable) {
```

Comment:
I think it makes more sense to use  Optional<String> instead of exception handling for control flows (that is a bad idea).

Reply:
Essentially just replacing the `throw ...` with `return Optional.empty()`?

Reply:
Yes, and Optional.of..(...) in the if block. I added some code suggestions

Reply:
Sure, I'll take a look at it. Thank you for the suggestion!
In theory, I like exceptions, but it does not make the as readable 😄

Reply:
Effective Java, Item 69: 
https://ahdak.github.io/blog/effective-java-part-9/
>Exceptions are, as their name implies, to be used only for exceptional conditions; they should never be used for ordinary control flow. A well designed API must not force its clients to use exceptions for ordinary control flow.